### PR TITLE
PPS pixel bug fix in error filling

### PR DESCRIPTION
**Pull request by fabferro**
#### PR description:

This PR is a bug fix in filling the persistent information about the errors coming from PPS Pixels during the data unpacking.
Data unpacking is not affected, only the way errors are decoded.
This PR is needed by a forthcoming PR to update the PPS Pixel DQM in order to display errors.

#### PR validation:

Validation has been done reprocessing Run-3 data (AlCaPPS data of run 355921, ~1M events) and running WF 136.892 of RunThematrix. 
The tests reported no changes in unpacked data, a part from the desired new error encoding.

No backport needed.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by fabferro** [comment]
It seems relvals ar efailing because of unreachable input files.

**Comment by francescobrivio** [comment]
> It seems relvals are failing because of unreachable input files.

Relval timeouts happen from time to time. Let's try again!

**Comment by francescobrivio** [comment]
type bugfix

**Comment by francescobrivio** [comment]
+alca

 - from the `alca` point of view it's just the addition of a simple method
 - tests are clean

**Comment by clacaputo** [comment]
+reconstruction

File: EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc [hunk 1]
```diff
@@ -124,63 +124,75 @@ void CTPPSPixelDataFormatter::interpretRawData(
...
+      }
+
       auto rawId = rocp.rawId();
 
       detDigis = &digis.find_or_insert(rawId);
       if ((*detDigis).empty())
         (*detDigis).data.reserve(32);  // avoid the first relocations
     }
 
+    if (skipROC || rocp.rawId() == 0)
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc:193]: @fabferro , just for checking: is it correct that `skipROC` is defined outside the loop, i.e. it remains set at the value taken during the previous iterations in the loop over `word`'s?

Comment (fabferro) [reply EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc:193]: Thanks Andrea for the question. The use of this flag is very delicate. The way it's assumed to work is that if it happens to be set true (i.e. a roc chip behaving badly) all the words from that roc are skipped. The flag can change inside the loop when the roc number changes (see lines 156-162) and the new is roc checked.

**Comment by perrotta** [comment]
+1

**Pull request merged**